CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_CIFacilityUpgrades.uc

```diff
@@ -0,0 +1,85 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  ArcaneData
+//  PURPOSE: Definitions for barracks size unlocks.
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+class X2StrategyElement_CIFacilityUpgrades extends X2StrategyElement_DefaultFacilityUpgrades;
+
+static function array<X2DataTemplate> CreateTemplates()
+{
+	local array<X2DataTemplate> Upgrades;
+
+	Upgrades.AddItem(CreateLivingQuarters_BarracksSizeI());
+	Upgrades.AddItem(CreateLivingQuarters_BarracksSizeII());
+	
+	return Upgrades;
+}
+
+static function X2DataTemplate CreateLivingQuarters_BarracksSizeI()
+{
+	local X2FacilityUpgradeTemplate Template;
+	local ArtifactCost Resources;
+
+	`CREATE_X2TEMPLATE(class'X2FacilityUpgradeTemplate', Template, 'LivingQuarters_BarracksSizeI');
+	Template.PointsToComplete = 0;
+	Template.MaxBuild = 1;
+	Template.strImage = "img:///UILibrary_StrategyImages.FacilityIcons.ChooseFacility_PowerConduitUpgrade";
+	Template.OnUpgradeAddedFn = OnUpgradeAdded_IncreaseBarracksSizeI;
+
+	// Stats
+	Template.iPower = 3;
+	Template.UpkeepCost = 10;
+
+	// Cost
+	Resources.ItemTemplateName = 'Supplies';
+	Resources.Quantity = 100;
+	Template.Cost.ResourceCosts.AddItem(Resources);
+
+	Resources.ItemTemplateName = 'EleriumDust';
+	Resources.Quantity = 10;
+	Template.Cost.ResourceCosts.AddItem(Resources);
```

Comment:
You got the costs already in config. No need to put them here, please remove (same with other upgrade)